pylint/checkers/typecheck.py

```diff
@@ -1029,10 +1029,12 @@ def visit_attribute(self, node: nodes.Attribute) -> None:
                 if not [
                     n
                     for n in owner.getattr(node.attrname)
-                    if not isinstance(n.statement(), nodes.AugAssign)
+                    if not isinstance(n.statement(future=True), nodes.AugAssign)
                 ]:
                     missingattr.add((owner, name))
                     continue
+            except exceptions.StatementMissing:
+                continue
```

Comment:
After the discussion about not adding the `FrameMissing` error in astroid, I've been thinking if we should deprecate / remove the `StatementMissing` error, too.

Is there a benefit here with it being more specific? I would probably recommend to just catch `ParentMissingError` here.

Reply:
I'm not sure if there is any benefit, perhaps when doing something like `[(i.frame(), j.statement()) for i,j in zip(nodes, other_nodes)]` but that is really really convoluted...
On the other hand, we have already gone to the effort of adding it and I don't think it adds too much boilerplate to the `astroid` codebase. We never know when we might need it in the future so I'd say let's just keep it in?

Reply:
Ok, let's keep it. Only because it's already in `astroid` 😉